[CI] MMROperatorTests testSimpleSmallInput failing

**Build Scans:**
- [elasticsearch-intake #34600 / part3](https://gradle-enterprise.elastic.co/s/kzxpdlzvhz35k)
- [elasticsearch-periodic-platform-support #11854 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/w3upb5c6f7bhe)
- [elasticsearch-periodic-platform-support #11854 / almalinux-8-aarch64_checkpart3_platform-support-arm](https://gradle-enterprise.elastic.co/s/22ceotepsxcze)
- [elasticsearch-periodic #11877 / openjdk21_checkpart3_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/4f4flcywk5mu2)
- [elasticsearch-periodic #11877 / openjdk21_checkpart3_java-fips-140-3-matrix](https://gradle-enterprise.elastic.co/s/czt5gfjrkj64e)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.operator.MMROperatorTests.testSimpleSmallInput" -Dtests.seed=C23275546E19A2DC -Dtests.locale=ky-Cyrl-KG -Dtests.timezone=Zulu -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.operator.MMROperatorTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSimpleSmallInput'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<1> but was:<2>
```

**Issue Reasons:**
- [main] 2 consecutive failures in step almalinux-8-aarch64_checkpart3_platform-support-arm
- [main] 2 consecutive failures in step openjdk21_checkpart3_java-fips-140-3-matrix
- [main] 5 failures in test testSimpleSmallInput (13.5% fail rate in 37 executions)
- [main] 2 failures in step almalinux-8-aarch64_checkpart3_platform-support-arm (100.0% fail rate in 2 executions)
- [main] 2 failures in step openjdk21_checkpart3_java-fips-140-3-matrix (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -315,15 +315,9 @@ tests:
 - class: org.elasticsearch.compute.lucene.read.ValuesSourceReaderOperatorTests
   method: testWithNulls
   issue: https://github.com/elastic/elasticsearch/issues/142434
-- class: org.elasticsearch.compute.operator.MMROperatorTests
-  method: testSimpleLargeInput
-  issue: https://github.com/elastic/elasticsearch/issues/142437
 - class: org.elasticsearch.xpack.esql.common.spatial.H3SphericalGeometryTests
   method: testIndexPoints
   issue: https://github.com/elastic/elasticsearch/issues/142439
-- class: org.elasticsearch.compute.operator.MMROperatorTests
-  method: testSimpleSmallInput
-  issue: https://github.com/elastic/elasticsearch/issues/142436
 - class: org.elasticsearch.xpack.esql.core.util.RemoteClusterUtilsTests
   method: testQualifyAndJoinIndices
   issue: https://github.com/elastic/elasticsearch/issues/142442

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MMROperatorTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/MMROperatorTests.java
@@ -22,9 +22,6 @@ import java.util.List;
 import static org.hamcrest.Matchers.equalTo;
 
 public class MMROperatorTests extends OperatorTestCase {
-
-    public static double NULL_BLOCK_CHANCE = 0.05;
-
     static List<float[]> TEST_VECTORS = List.of(
         new float[] { 0.4f, 0.2f, 0.4f, 0.4f },
         new float[] { 0.4f, 0.2f, 0.3f, 0.3f },
@@ -53,12 +50,8 @@ public class MMROperatorTests extends OperatorTestCase {
             @Override
             protected Page createPage(int positionOffset, int length) {
                 length = Integer.min(length, remaining());
+                currentPosition += length;
                 var blocks = new Block[1];
-
-                if (randomDouble() < NULL_BLOCK_CHANCE) {
-                    blocks[0] = blockFactory.newConstantNullBlock(length);
-                    return new Page(blocks);
-                }
 
                 float[] vectors = new float[length * 4];
                 int[] vectorPositions = new int[length + 1];
@@ -77,7 +70,7 @@ public class MMROperatorTests extends OperatorTestCase {
                     blocks[0] = blockFactory.newFloatArrayBlock(vectors, length, vectorPositions, new BitSet(), Block.MvOrdering.UNORDERED);
                 } catch (Exception e) {
                     throw new RuntimeException(e);
-                } finally {}
+                }
 
                 finish();
                 return new Page(blocks);

